mrc/localization/locator.py

```diff
@@ -1,36 +1,41 @@
 import threading
-from dynaconf import settings
-import itertools
 import time
+from typing import Callable
 
 
 class Locator(threading.Thread):
-    def __init__(self):
+    def __init__(self, read_data: Callable, process_data: Callable, interval: float):
```

Comment:
How about Observable instead of Callable and IoC this way?
https://pyrx.sourceforge.io/